src/main/webapp/app/exercises/file-upload/manage/file-upload-exercise-detail.component.ts

```diff
@@ -59,6 +61,7 @@ export class FileUploadExerciseDetailComponent implements OnInit, OnDestroy {
                 (fileUploadExerciseResponse: HttpResponse<FileUploadExercise>) => {
                     this.fileUploadExercise = fileUploadExerciseResponse.body!;
                     this.isExamExercise = this.fileUploadExercise.exerciseGroup !== undefined;
+                    this.course = this.isExamExercise ? this.fileUploadExercise.exerciseGroup?.exam?.course : this.fileUploadExercise.course;
```

Comment:
There is already a helper function for that: https://github.com/ls1intum/Artemis/blob/develop/src/main/webapp/app/entities/exercise.model.ts#L194-L196

I think it would be better to use it